src/main/kotlin/app/revanced/patcher/fingerprint/method/impl/MethodFingerprint.kt

```diff
@@ -42,6 +44,93 @@ abstract class MethodFingerprint(
     var result: MethodFingerprintResult? = null
 
     companion object {
+        private fun MethodFingerprint.signatureKey(): String {
+            if (accessFlags == null) return ""
+
+            var returnTypeValue = returnType
+            if (returnTypeValue == null) {
+                if (accessFlags and AccessFlags.CONSTRUCTOR.value != 0) {
+                    // Constructors always have void return type
+                    returnTypeValue = "V"
+                } else {
+                    return ""
+                }
+            }
+
+            return buildString {
+                append(accessFlags)
+                append(returnTypeValue.first())
+                parameters?.forEach { append(it.first()) }
+            }
+        }
+
+        internal fun createMethodLookupMap(classes: Iterable<ClassDef>) : Map<String, List<Pair<ClassDef, Method>>> {
```

Comment:
The return type is not readable. It is not apparent by directly looking at the return value what this method is returning. Create classes and/or typealiases